### PR TITLE
handle error scenarios in attestation api

### PR DESCRIPTION
**Pull request by AnieeG**
## Solution

### CODE REVIEW

File: core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go [hunk 1]
```diff
@@ -139,18 +140,20 @@ func (s *TokenDataReader) ReadTokenData(ctx context.Context, msg internal.EVM2EV
...
-
-	if attestationResp.Status != attestationStatusSuccess {
-		return []byte{}, tokendata.ErrNotReady
-	}
-
-	// The USDC pool needs a combination of the message body and the attestation
-	messageAndAttestation, err = encodeMessageAndAttestation(messageBody, attestationResp.Attestation)
-	if err != nil {
-		return nil, fmt.Errorf("failed to encode messageAndAttestation : %w", err)
+	switch attestationResp.Status {
```
Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go:147]: Can we have a test for that?

**Review by mateusz-sekara — APPROVED** [review]

File: core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go [hunk 2]
```diff
@@ -44,6 +44,10 @@ const (
 	attestationStatusPending attestationStatus = "pending_confirmations"
 )
 
+var (
+	ErrAttestionAPI = errors.New("unexpected response from attestation API")
```
Comment (dimkouv) [inline core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go]: maybe rename? `ErrUnknownResponse` ?

File: core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go [hunk 3]
```diff
@@ -139,18 +144,20 @@ func (s *TokenDataReader) ReadTokenData(ctx context.Context, msg internal.EVM2EV
...
+	switch attestationResp.Status {
+	case attestationStatusSuccess:
+		// The USDC pool needs a combination of the message body and the attestation
+		messageAndAttestation, err = encodeMessageAndAttestation(messageBody, attestationResp.Attestation)
+		if err != nil {
+			return nil, fmt.Errorf("failed to encode messageAndAttestation : %w", err)
+		}
+		return messageAndAttestation, nil
+	case attestationStatusPending:
+		return nil, tokendata.ErrNotReady
```
Comment (dimkouv) [inline core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go:156]: can we include [here](https://github.com/smartcontractkit/ccip/blob/ee7e13aac99729bff32c37832ad2647636fe4685/core/services/ocr2/plugins/ccip/ccipexec/ocr2.go#L400) some handling for `tokendata.ErrNotReady` ? I see it's not used anywhere

```
maybe:
if err != nil {
   if errors.Is(err, tokendata.ErrNotReady) {
      msgLggr.Warnw("skipping message token data not ready", err)
   } else {
      msgLggr.Errorw("skipping message error while getting token data", "err", err) 
   }
   continue
}
```

File: core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go (hunk 2 above)
Comment (AnieeG) [reply core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go]: updated

File: core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go (hunk 3 above)
Comment (AnieeG) [reply core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go:156]: Added this

**Pull request merged**